### DISCUSSION TITLE
Renogy Rego 400ah compatibility anyone?

### DISCUSSION THREAD

**Discussion by yarrumretep** [Louisvdw/dbus-serialbattery#1069] [General]
Wondering if anyone has gotten the dbus-serialbattery driver to work with the Renogy REGO 400ah batteries?  I have a system with three of those in parallel and I'd like to get them talking to my CERBO-GX.  They do speak bluetooth - is there a way to get the bluetooth connection to work?

**Comment by yarrumretep** [comment]
The REGOs also have some flavor of CAN bus - but the CERBO-gx doesn't seem to recognize it under any of the VECAN or BMS CAN settings...

Seems like we ought to be able to crack this... I had thought Renogy compatibility would cover me - but I guess the REGO stuff is totally diff...

**Comment by yarrumretep** [comment]
OK - I've solved this.  It turns out the Renogy Batteries speak "RC-V" CAN bus protocol.  With a custom re-wiring of the RJ-45 on the connector for the Renogy One display (using the Cerbo-gx documentation for the pinouts on the VE.CAN and the standardized wire color scheme from the RC-V CAN protocol), I was able to plug this into the VE.CAN bus on the Cerbo-gx and set that bus to use the RC-V protocol.   From there it apparently "just worked" - it even seemed to merge the three Renogy batteries I have into one battery bank!

While this solution is not using dbus-serialbattery at all, I thought it might be helpful if others run into this problem out in the wild.

**Reply by mr-manuel** [reply]
Thanks for sharing your solution with others! Maybe also a picture of the cable could help and/or a wiring diagram.

**Reply by rogergrant99** [reply]
I also have a rego 400ah battery connected and I came to the same conclusion, the RV-C bus selection worked, but it does not show individual cell voltages which dbus-serialbattery does, is there a way to get dbus-serialbattery to talk to a RV-C BMS battery ?

**Reply by TheVodden** [reply]
It appears that you would need to do a protocol conversion from RV-C to Modbus.  Looking at the specs, The bluetooth connection is Modbus, but there is no bluetooth connectivity between Renogy and the driver.

**Reply by rogergrant99** [reply]
My Rego 400AH has no Bluetooth unfortunately, I believe it DOA on the battery purchase. My only option is the CAN-BUS. So would there be interest in writing a RV-C driver ?

**Reply by TheVodden** [reply]
Do we know if the Rego has data available for the info you're wanting to retrieve?  Is there documentation on this?

**Reply by rogergrant99** [reply]
I will try to find out.

**Reply by Quadron** [reply]
Great to hear you got that working, I have RBT100LFP12S batteries, but a rego DC-DC charger that I want to wire into the GX via the RV-C bus. Recently ordered a rego to rj45 cable, but have not had the time to test it yet.
The batteries are my priority, but have had no luck whatsoever with that for over a year... 
Could you post your pin to pin connections for the rego to ve.can?

**Reply by TheVodden** [reply]
@Quadron you need to check out [this](https://github.com/Louisvdw/dbus-serialbattery/issues/142#issuecomment-2185323434) thread.  This is now working. @mr-manuel debugged this on my system and it is currently working flawlessly.  I too have RBT100LFP12S (4) connected to the Cerbo with a single RS485 connection.  I am using [this](https://a.co/d/0akxqsim) adapter.  If it works for you please be sure to donate, as he spent considerable time making this work.

**Reply by TheVodden** [reply]
@rogergrant99 I looked through the RV-C documentation and sadly it appears that the data you want to retrieve isn't available.

**Reply by rogergrant99** [reply]
Thanks for looking. The current information displayed is adequate. It would have been nice to get more details.

**Comment by BlueY21** [comment]
Does anyone know if a VE.CAN bus connection to a Cerbo GX using RC-V will
work for a bank of three paralleled Renogy Smart Lithium Iron Phosphate
(RBT100LFP12S - V2) batteries?

The RS485 LINK and UP Communication Ports of my three batteries are
daisy-chained and the first battery is also connected to a Renogy Bluetooth
transmitter that reports to a phone app.  Thus, the Renogy BMS manages the
batteries as one and the phone app reports the status of all and each.

I connected the first one thru a RS485 splitter and VE USB - RS485 cable
also to my Cerbo with dbus-serialbattery driver.  It seems to work,
reporting the status of that first battery.  The Cerbo disconnects briefly
when the Cerbo and BT module communications collide, but that's not a
big deal.  I would prefer not to have to add two more RS485 splitters and
VE USB - RS485 cables and a USB bridge to connect then aggregate the
remaining batteries.
Also, yarrumretep, a wiring diagram and/or picture of your cable would be
great.

**Reply by TheVodden** [reply]
I would also be interested in this.  I have 4 RBT100LFP12S V3 in parallel and would like to poll data from these into the Cerbo GX,  I haven't had any luck at all bringing in any data other than the model number into the Cerbo with the driver.  They are also daisy chained together in the same manner as yours, with the exception of a renogy battery monitor instead of a BT2.  I haven't installed a splitter, instead removing the battery monitor to connect via 485.  

I did find the modbus manual for these batteries, and it shows the pinout for both CAN and RS485 on the battery end.  Worth noting here that the Renogy pin numbers are backwards to conventional pin numbers, showing 1 as 8 and 8 as 1.  It lists CAN-H as 6 (3), CAN-L as 8 (1), and GND as pin 4 (5) with a baud rate of 500kbps.  I also found that VE.Can is CAN-H on pin 7, CAN-L on Pin 8, and GND on pin 3.  This should assist in cable making.  I'm going to give this a try and see what I get.  

[Lithium.Iron.Battery.BMS.Modbus.Protocol.V1.7.zh-CN.en.1.pdf](https://github.com/user-attachments/files/15880835/Lithium.Iron.Battery.BMS.Modbus.Protocol.V1.7.zh-CN.en.1.pdf)

**Reply by TheVodden** [reply]
A follow up to this.  

It looks to me from the battery documentation that the RBT100LFP12S batteries are RS485 only, with no CAN.

I poked around a bit with qModMaster and found that my batteries are addressed from 48 to 51, but the driver can only read one battery.

**Reply by Quadron** [reply]
Thankfully someone finally can confirm that at least something works for them!
For over a year now I've been trying to get my 4 daisychained RBT100LFP12S batteries to talk to the GX via RS485. I've tried the pins as in the renogy documentation and in reverse as stated in various locations. Nothing. I even tried connecting every possible pin combination for A an B wires with a victron RS485 cable. Still nothing. I see the driver polling the battery as per instructions of the troubleshooting, but never get any response. I gave up for a while and just used the app.
Tried a chinesium rs485 to RJ45 cable today, still nothing. 
What are the pins you are using to get at least the first battery to talk to the GX?

**Comment by TheVodden** [comment]
Pinout by 'conventional' numbering is 6 and 7, or if you are using an ethernet cable, green and striped brown.  A would be 7 and B would be 6.  However, Renogy numbers the pins backwards, so they say they are pin 2 and 3 as they number the connector backwards.  

Just remember brown/white and green. 

The attached image is from a Renogy charge controller, not the battery, so it shows CAN pins also (which the battery doesn't have), but the 485 pins are illustrated correctly.

![Renogy Pinout](https://github.com/Louisvdw/dbus-serialbattery/assets/173129376/24b79c43-d120-453e-b218-e22b3f04ec3c)

**Reply by TheVodden** [reply]
with that being said, if you had a Renogy cable with an RJ45 end from the Rego, it should have the CAN pinout the same as above on 8 and 6 (actually 1 (CAN L) and 3 (CAN H) in the real world).  

VeCan appears to use pins 7 (CAN H) and 8 (CAN L), so an adapter cable crossing those pins over, or recrimping the Renogy cable to wire direct should work.  
![VeCAN](https://github.com/Louisvdw/dbus-serialbattery/assets/173129376/925de2d9-502a-4f79-b550-4638ae1437c8)

**Reply by Quadron** [reply]
Ok, Thank you. I had tried every single combination, without success, but now it seems to work. I've gotten the first battery talking to the GX. Now I'm figuring out the addresses to use for the other 3 with the Dbus-serialbattery for multiple batteries on one adapter.

**Reply by ispookie666** [reply]
Hi
I'm sorry to bother.  I'm having no luck with this.  
Per Victron the pin out for their CAN -
GND - Pin 3 (Green- White)
Can H - Pin 8 (Brown)
Can L - Pin 7 (White-Brown)
I understand the renogy pins are the other way around, but correcting for this I have had no luck.  Any help would be appreciated.  For the Victron end I used B-type wiring

**Reply by nshawdvm-cpu** [reply]
When you do this does the Renogy then communicate charge parameters to CerboGX and then on the a Victron inverter? Or does it just display battery info?

**Reply by huanvu** [reply]
@ispookie666 I believe you have Can H and Can L mixed around

**Comment by Quadron** [comment]
When you do this, the battery communicates battery state of charge, voltage and current (charge or discharge rate) to the gx.
Serialbattery then calculates charge settings for the battery based on the configuration of serialbattery and the controls the connected chargers. They will report "external control" in the gx screen.

Vital is you set the correct settings for the battery in the serialbattery configuration.

Connect the chargers and give it some time. In my experience It can take a few minutes before charging really starts, especially with the multiplus.

I have 4 in parallel with the canbus cables daisychained and it has worked great.
I have not gotten around to implement a battery aggregator but have assigned a single battery as the leading one. That has worked ok up to now. But keep checking your own setup!

Sent from [Proton Mail](https://proton.me/mail/home) for Android.

-------- Original Message --------

**Comment by huanvu** [comment]
does anyone know the pinout on the LP16 connection end?

**Reply by huanvu** [reply]
LP16 pinout to RJ45 T568B
CAN H - LP16 Pin 5 - RJ45 Brown-White(Pin 7)
CAN L - LP16 Pin 7 - RJ45 Brown (Pin 8)
GROUND - LP16 Pin 4 on some models, Pin 2 on others - RJ45 Green-White (Pin 3)

pins should be marked on the connector itself, don't follow renogy documentation unless you want to be confused.

**Comment by yarrumretep** [comment]
Also check out https://github.com/yarrumretep/dbus-rvc-renogy for REGO - Cerbo connectivity restoration for recent Cerbo software version.

**Reply by huanvu** [reply]
Thank you! Excited to try this out.
